driver/src/contracts/stablex_auction_element.rs

```diff
@@ -24,6 +26,16 @@ impl StableXAuctionElement {
     /// Sets `id` to `0` because this information is not contained in the
     /// serialized information.
     pub fn from_bytes(bytes: &[u8; AUCTION_ELEMENT_WIDTH]) -> Self {
+        let mut indexed_bytes = [0u8; INDEXED_AUCTION_ELEMENT_WIDTH];
+        for (index, bit) in bytes.iter().enumerate() {
```

Comment:
nit: Can you use copy_from_slice here?
```suggestion
        indexed_bytes[..AUCTION_ELEMENT_WIDTH].copy_from_slice(&bytes)
```

Reply:
I wanted to but can't because of https://doc.rust-lang.org/std/primitive.slice.html#panics-17

> This function will panic if the two slices have different lengths.

Reply:
Yes, but that is why you first slice `indexed_bytes` to `AUCTION_ELEMENT_WIDTH` length, and then use `copy_from_slice`.

Reply:
I'm copying bytes into indexed_bytes, so I'm looking for a way to extend bytes with 2 zeros (actually making it larger not small). I agree that the current version doesn't feel syntactically as sweetest as it can be, but I don't think I can slice an array up rather than down can I?

Reply:
Sorry, I didn't explain it well :disappointed:. This is what I meant: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=b0081941a120670bf8a018cc95087efd

Reply:
Ah cool, that's what I was missing. Let me adjust this in a separate PR.